src/openforms/authentication/contrib/digid_eherkenning_oidc/plugin.py

```diff
@@ -179,8 +179,25 @@ def transform_claims(self, normalized_claims: DigiDClaims) -> FormAuth:
         }
 
 
+class EHClaims(TypedDict):
+    """
+    Processed EH claims structure.
+
+    See :attr:`digid_eherkenning.oidc.models.EHerkenningConfig.CLAIMS_CONFIGURATION`
+    for the source of this structure.
+    """
+
+    identifier_type_claim: NotRequired[str]
+    legal_subject_claim: str
+    acting_subject_claim: str
+    branch_number_claim: NotRequired[str]
+    # *could* be a number if no value mapping is specified and the source claims return
+    # numeric values...
+    loa_claim: NotRequired[str | int | float]
+
+
 @register("eherkenning_oidc")
-class eHerkenningOIDCAuthentication(OIDCAuthentication):
+class eHerkenningOIDCAuthentication(OIDCAuthentication[EHClaims]):
```

Comment:
ah yeah, it's a typing generic. The base class defines/uses the typevar `T`, which is used again in the `transform_claims` method. This way, we can specify and type-check the expected shape of the claims for each plugin, so that we don't acctidentally access keys that don't exist or fail to handle them being optional.